numpy/ma/core.py

```diff
@@ -1430,7 +1427,7 @@ def getmask(a):
 get_mask = getmask
 
 
-def getmaskarray(arr):
+def getmaskarray(arr, nomask_as_readonly=False):
```

Comment:
My own sense would be not to add an argument here, which is used by `_viewmaskarray`, but rather have this call `_viewmaskarray` with a new argument; that way the rest of the code is not slowed down by another function call.

I'd also suggest to make the argument `writable=True`, ie., it states what kind of view is required.